### DISCUSSION TITLE
Suggested improvements ...

### DISCUSSION THREAD

**Discussion by darth2602** [lllyasviel/Fooocus#2460] [Ideas]
Note that this Colab will disable refiner by default because Colab free's resources are relatively limited (and some "big" features like image prompt may cause free-tier Colab to disconnect). We make sure that basic text-to-image is always working on free-tier Colab.

> It would be interesting to: 
- be able to manually add a few lines to activate Refiner when you have Colab Pro and make the most of its resources   
- be able to save our images in google drive 
- add an upscaler like 4x ultrasharp - add "Dynamic Prompt".

**Comment by mashb1t** [comment]
Can you please elaborate what you mean by "Dynamic Prompt"?

**Reply by darth2602** [reply]
Yes , an exemple (it' not the exact syntax) 

A beautiful {black|White|brown} Cat with {Blue|Yellow} Eyes   
->   image 1 : A beautiful Black Cat With blue Eyes  
-> image 2 : A beautiful Bron Cat With yellow Eyes
->image 3 : A beautiful white Cat With blue Eyes    

https://github.com/adieyal/sd-dynamic-prompts?tab=readme-ov-file

**Reply by patrickchamp** [reply]
You can now do this with arrays as of last week’s update. Format is [black, white, brown].

if it’s wildcards collections that you want, you can create a txt file in the wildcards folder like catcolor.txt with each colour on a separate line. In your prompt, add`__catcolor__ `

Arrays are selected in order. Wildcards are selected randomly.

**Comment by mashb1t** [comment]
@darth2602 for your first point, refiner in colab, you can simply download and use any sdxl or even sd1.5 model as refiner, see https://github.com/lllyasviel/Fooocus/issues/1922 (e.g. --preset sai, check out the sai config).
It is recommended to set up a preset yourself, maybe apply the rebalancing of https://github.com/lllyasviel/Fooocus/pull/1710, and choose the refiner then by default.

**Comment by darth2602** [comment]
ha oui....j'oubliait .... a night mode for the interface would be nice.

**Reply by mashb1t** [reply]
This is already included, feel free to set your OS / browser to dark mode and check again
